common/headers/caller_info.go

```diff
@@ -33,47 +33,117 @@ import (
 const (
 	CallerTypeAPI        = "api"
 	CallerTypeBackground = "background"
+
+	CallerNameSystem = "system"
 )
 
-type CallerInfo struct {
-	CallerType string
+var (
+	SystemBackgroundCallerInfo CallerInfo = CallerInfo{
+		CallerName: CallerNameSystem,
+		CallerType: CallerTypeBackground,
+	}
+)
 
-	// TODO: add fields for CallerName and CallerInitiation
+type CallerInfo struct {
+	CallerName     string
+	CallerType     string
```

Comment:
add comments to indicate CallerName could be system or user namespace, caller type could be API or background.